samples/nrf9160/modem_shell/src/drivers/nrf91_non_offloaded_dev.c

```diff
@@ -0,0 +1,507 @@
+ *
+ */
+
+/* nRF91 usage without offloading TCP/IP stack to modem,
+ * i.e. using Zephyr native TCP/IP stack.
+ */
+
+#include <zephyr.h>
+
+#include <random/rand32.h>
+#include <net/dummy.h>
+#include <net/net_pkt.h>
+#include <net/net_if.h>
+#include <net/net_ip.h>
+
+#include <posix/arpa/inet.h>
+
+#include <nrf_socket.h>
+
+#include <modem/pdn.h>
+#include <nrf_modem_at.h>
+#include <modem/at_cmd_parser.h>
+#include <modem/at_params.h>
+
+#include "link/link_shell_pdn.h"
+#include "utils/net_utils.h"
+#include "mosh_print.h"
+
+#define NO_MDM_SCKT -1
+
+K_SEM_DEFINE(mdm_socket_sem, 0, 1);
+
+/* Work for creating modem data socket (uses system workq): */
+static struct k_work_delayable mdm_socket_work;
+
+struct nrf91_non_offload_dev_context {
+	uint8_t mac_addr[6];
+	struct net_if *iface;
+	bool default_pdp_active;
+	int mdm_skct_id;
+};
+
+struct nrf91_non_offload_dev_context nrf91_non_offload_iface_data;
+
+/******************************************************************************/
+
+static void util_get_ip_addr(int cid, char *addr4, char *addr6)
+{
+	int ret;
+	char cmd[128];
+	char tmp[sizeof(struct in6_addr)];
+	char addr1[NET_IPV6_ADDR_LEN] = { 0 };
+	char addr2[NET_IPV6_ADDR_LEN] = { 0 };
+
+	sprintf(cmd, "AT+CGPADDR=%d", cid);
+
+	/** parse +CGPADDR: <cid>,<PDP_addr_1>,<PDP_addr_2>
+	 * PDN type "IP": PDP_addr_1 is <IPv4>, max 16(INET_ADDRSTRLEN), '.' and digits
+	 * PDN type "IPV6": PDP_addr_1 is <IPv6>, max 46(INET6_ADDRSTRLEN),':', digits, 'A'~'F'
+	 * PDN type "IPV4V6": <IPv4>,<IPv6>, or <IPV4> or <IPv6>
+	 */
+	ret = nrf_modem_at_scanf(cmd, "+CGPADDR: %*d,\"%46[.:0-9A-F]\",\"%46[:0-9A-F]\"", addr1,
+				 addr2);
+	if (ret <= 0) {
+		return;
+	}
+
+	/* parse 1st IP string */
+	if (addr4 != NULL && inet_pton(AF_INET, addr1, tmp) == 1) {
+		strcpy(addr4, addr1);
+	} else if (addr6 != NULL && inet_pton(AF_INET6, addr1, tmp) == 1) {
+		strcpy(addr6, addr1);
+		return;
+	}
+
+	/* parse second IP string (IPv6 only) */
+	if (addr6 == NULL) {
+		return;
+	}
+	if (ret > 1 && inet_pton(AF_INET6, addr2, tmp) == 1) {
+		strcpy(addr6, addr2);
+	}
+}
+
+/******************************************************************************/
+
+struct pdn_conn_dyn_params {
+	struct in_addr dns_addr4_primary;
+	struct in_addr dns_addr4_secondary;
+	struct in6_addr dns_addr6_primary;
+	struct in6_addr dns_addr6_secondary;
+	uint32_t ipv4_mtu;
+	uint32_t ipv6_mtu;
+};
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO "AT+CGCONTRDP=%d"
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_PARAM_COUNT 20
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_DNS_ADDR_PRIMARY_INDEX 6
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_DNS_ADDR_SECONDARY_INDEX 7
+#define AT_CMD_PDP_CONTEXT_READ_DYN_INFO_MTU_INDEX 12
+
+static int util_get_pdn_conn_dyn_params(int cid, struct pdn_conn_dyn_params *ret_dyn_info)
+{
+	struct at_param_list param_list = { 0 };
+	size_t param_str_len;
+	char *next_param_str;
+	char at_cmd_str[16];
+	char at_cmd_response_str[1024];
+	char dns_addr_str[256];
```

Comment:
Ain't that a bit too much for a stack? I'd increase the sys workqueue stack requirements greatly.

Reply:
No actual need for such big reservations from stack. Halfed.